packages/grid/x-data-grid-pro/src/utils/domUtils.ts

```diff
@@ -28,31 +28,28 @@ export function findGridCellElementsFromCol(col: HTMLElement, api: GridPrivateAp
   const colIndex = Number(ariaColIndex) - 1;
   const cells: Element[] = [];
 
-  const virtualScrollerContent = api.virtualScrollerRef?.current?.firstElementChild;
-  if (!virtualScrollerContent) {
-    return [];
-  }
-
-  const renderedRowElements = virtualScrollerContent.querySelectorAll(
-    `:scope > div > .${gridClasses.row}`, // Use > to ignore rows from detail panels
-  );
-
-  renderedRowElements.forEach((rowElement) => {
-    const rowId = rowElement.getAttribute('data-id');
-    if (!rowId) {
-      return;
-    }
-
-    let columnIndex = colIndex;
-
-    const cellColSpanInfo = api.unstable_getCellColSpanInfo(rowId, colIndex);
-    if (cellColSpanInfo && cellColSpanInfo.spannedByColSpan) {
-      columnIndex = cellColSpanInfo.leftVisibleCellIndex;
-    }
-    const cell = rowElement.querySelector(`[data-colindex="${columnIndex}"]`);
-    if (cell) {
-      cells.push(cell);
-    }
+  api.virtualScrollerRef?.current?.childNodes.forEach((child) => {
```

Comment:
A small opportunity: We could possibly avoid one loop by targeting rows directly from the `virtualScroller`.
```diff
diff --git a/packages/grid/x-data-grid-pro/src/utils/domUtils.ts b/packages/grid/x-data-grid-pro/src/utils/domUtils.ts
index 42046bf72..b21304b32 100644
--- a/packages/grid/x-data-grid-pro/src/utils/domUtils.ts
+++ b/packages/grid/x-data-grid-pro/src/utils/domUtils.ts
@@ -28,9 +28,9 @@ export function findGridCellElementsFromCol(col: HTMLElement, api: GridPrivateAp
   const colIndex = Number(ariaColIndex) - 1;
   const cells: Element[] = [];
 
-  api.virtualScrollerRef?.current?.childNodes.forEach((child) => {
-    const renderedRowElements = (child as HTMLElement).querySelectorAll(
-      `:scope > div > .${gridClasses.row}`, // Use > to ignore rows from detail panels
+  if (api.virtualScrollerRef?.current) {
+    const renderedRowElements = api.virtualScrollerRef.current.querySelectorAll(
+      `div > .${gridClasses.row}`, // Use > to ignore rows from detail panels
     );
 
     renderedRowElements.forEach((rowElement) => {
@@ -50,7 +50,7 @@ export function findGridCellElementsFromCol(col: HTMLElement, api: GridPrivateAp
         cells.push(cell);
       }
     });
-  });
+  }
 
   return cells;
 }
```

Reply:
This will break https://github.com/mui/mui-x/pull/9670 (the very same PR that caused this regression 😅)

Reply:
Ah, correct 🤦  I missed backtracking the original problem.

Just out of curiosity I tried updating the selector from `div > .${gridClasses.row}` to `:scope > .${gridClasses.virtualScrollerContent} > div > .${gridClasses.row}` and it seems to work without breaking #9670.
Its a CSS-only solution, still I am not sure if it's the most elegant of the solutions, since the selector has become a bit longer than usual. 😆 
Feel free to go with whatever you feel better.

Reply:
Ah, correct 🤦 I missed backtracking the original problem.

Just out of curiosity I tried updating the selector from `div > .${gridClasses.row}` to `:scope > div > div > .${gridClasses.row}` and it seems to work without breaking https://github.com/mui/mui-x/pull/9670.
Its a CSS-only solution, still I am not sure if it's the most elegant of the solutions, since the selector has become a bit longer than usual and also depends on a strict DOM hierarchy. 😆
Feel free to go with whatever you feel better.

Reply:
Also, comment `// Use > to ignore rows from detail panels` could probably become `// Use > to ignore rows from nested data grids` as the grid could also be nested in a cell other than a detail panel.

Reply:
Alright, that works, thanks!